lang/cmu_is_lex/cmu_is_lex.c

```diff
@@ -0,0 +1,316 @@
+    for (i=0; cmu_is_lex_onset_trigrams[i]; i++)
+        if (cst_streq(x,cmu_is_lex_onset_trigrams[i]))
+            return TRUE;
+    return FALSE;
+}
+
+int cmu_is_syl_boundary_mo(const cst_item *i,const cst_val *rest)
+{
+    // TODO adapt to is
+    /* syl boundary maximal onset */
+    int d2v;
+
+    if (rest == NULL)
+	return TRUE;
+    else if (cmu_is_is_silence(val_string(val_car(rest))))
+	return TRUE;
+    else if (!cmu_is_has_vowel_in_list(rest)) 
+        /* no more vowels so rest *all* coda */
+	return FALSE;
+    else if (!cmu_is_has_vowel_in_syl(i))  /* need a vowel */
+        /* no vowel yet in syl so keep copying */
+	return FALSE;
+    else if (cmu_is_is_vowel(val_string(val_car(rest))))
+        /* next is a vowel, syl has vowel, so this is a break */
+	return TRUE;
+    else if (cst_streq("ng",val_string(val_car(rest))))
+        /* next is "ng" which can't start a word internal syl */
+	return FALSE;
+    else 
+    {
+        /* want to know if from rest to the next vowel is a valid onset */
+        d2v = cmu_is_lex_dist_to_vowel(rest);
+        if (d2v < 2)
+            return TRUE;
+        else if (d2v > 3)
+            return FALSE;
+        else if (d2v == 2) 
+            return cmu_is_lex_onset_bigram(rest);
+        else /* if (d2v == 3) */
+            return cmu_is_lex_onset_trigram(rest);
+        return TRUE;
+    }
+
+}
+
+cst_utterance *cmu_is_lex_postlex(cst_utterance *u)
+{
+    return u;
+}
+
+cst_lexicon cmu_is_lex;
+cst_lts_rules cmu_is_lts_rules;
+extern const char * const cmu_is_lts_phone_table[];
+extern const char * const cmu_is_lts_letter_table[];
+extern const cst_lts_addr cmu_is_lts_letter_index[];
+extern const cst_lts_model cmu_is_lts_model[];
+
+cst_lexicon *cmu_is_lex_init(void)
+{
+    /* Should it be global const or dynamic */
+    /* Can make lts_rules just a cart tree like others */
+
+    if (cmu_is_lts_rules.name)
+    return &cmu_is_lex;  /* Already initialized */
+
+    cmu_is_lts_rules.name = "cmu_is";
+    cmu_is_lts_rules.letter_index = cmu_is_lts_letter_index;
+
+    cmu_is_lts_rules.models = cmu_is_lts_model;
+
+    cmu_is_lts_rules.phone_table = cmu_is_lts_phone_table;
+    cmu_is_lts_rules.context_window_size = 4;
+    cmu_is_lts_rules.context_extra_feats = 1;
+    cmu_is_lts_rules.letter_table = 0 /* cmu_is_lts_letter_table */;
+
+    cmu_is_lex.name = "cmu_is";
+    cmu_is_lex.num_entries = cmu_is_lex_num_entries;
+
+    cmu_is_lex.data = (unsigned char *)(void *)cmu_is_lex_data;
+
+    cmu_is_lex.num_bytes = cmu_is_lex_num_bytes;
+    cmu_is_lex.phone_table = (char **) cmu_is_lex_phone_table;
+    cmu_is_lex.syl_boundary = cmu_is_syl_boundary_mo;
+    cmu_is_lex.lts_rule_set = (cst_lts_rules *) &cmu_is_lts_rules;
+
+    cmu_is_lex.phone_hufftable = cmu_is_lex_phones_huff_table;
+    cmu_is_lex.entry_hufftable = cmu_is_lex_entries_huff_table;
+
+    cmu_is_lex.postlex = cmu_is_lex_postlex;
+
+    return &cmu_is_lex;
+
+    // cst_lexicon *l;
```

Comment:
Mind to comment the commented out code ? Or rather: if we never need it: remove ?